core/conanfile.py

```diff
@@ -64,14 +62,10 @@ def requirements(self):
             else:
                 self.requires("mpdecimal/2.5.0")
 
-        # The pre-conan-center-index version of `openssl` was capitalized as `OpenSSL`.
-        # Both versions can't live in the same Conan cache so we need this compatibility
-        # option to pick the available version. The cache case-sensitivity issue should
-        # be solved in Conan 2.0, but we need this for now.
-        if self.options.openssl_variant == "lowercase":
-            self.requires("openssl/1.1.1k")
+        if self.pyversion >= scm.Version("3.10.0"):
+            self.requires("openssl/3.1.1")
```

Comment:
`openssl/3.1.1` [crashes](https://github.com/openssl/openssl/issues/20753) on MacOS arm when running with a debugger attached, the fix has been [merged](https://github.com/openssl/openssl/pull/21261) but it's not released yet. I'd recommend either going for `3.0.8` or waiting for `3.1.2`

Reply:
Great catch! I've change the version and added a comment to reflect this.